roboteam_ai/src/coach/defence/DefencePositionCoach.cpp

```diff
@@ -290,7 +290,10 @@ std::shared_ptr<Line> DefencePositionCoach::blockToGoalLine(const PossiblePass &
 /// searches the most dangerous position and then gets the segment which blocks that (if it exists/is possible)
 std::shared_ptr<Line> DefencePositionCoach::blockBallLine(const world::WorldData &simulatedWorld) {
     Vector2 mostDangerousPos = getMostDangerousPos(simulatedWorld);
-    return getBlockLineSegment(world::field->getGoalSides(true), mostDangerousPos);
+    if (world::field->pointIsInField(mostDangerousPos)){
+        return getBlockLineSegment(world::field->getGoalSides(true), mostDangerousPos);
```

Comment:
what is a blocklinesegment?

Reply:
read defenders documentation if you want to know more. t's a segment of a line on which you can stand to block towards the goal (hence BlockLineSegment)